test/test_client.py

```diff
@@ -1836,6 +1836,28 @@ def test_handshake_08_invalid_aws_ec2(self):
             None,
         )
 
+    def test_dict_hints(self):
+        c = rs_or_single_client()
```

Comment:
These tests should use the global client instead of creating a new one since creating clients is expensive and slows down the test suite.

Reply:
Good catch, opened https://github.com/mongodb/mongo-python-driver/pull/1396 to fix.